src/runtime/device_interface.cpp

```diff
@@ -222,17 +247,11 @@ WEAK int halide_device_malloc(void *user_context, struct halide_buffer_t *buf,
         return halide_error_code_incompatible_device_interface;
     }
 
-    // Ensure code is not freed prematurely.
-    // TODO: Exception safety...
-    device_interface->impl->use_module();
-    result = device_interface->impl->device_malloc(user_context, buf);
-    device_interface->impl->release_module();
-
-    if (result) {
+    if (auto result = call_device_interface(device_interface, device_interface->impl->device_malloc, user_context, buf); result != halide_error_code_success) {
         return halide_error_code_device_malloc_failed;
-    } else {
-        return 0;
     }
+    halide_debug_assert(user_context, buf->device == 0);
```

Comment:
I think this is meant to be in device_free

Reply:
Yikes, yes, I looked at the line numbers and not the call